src/benchmark/benchmark_activeobjectmgr.cpp

```diff
@@ -47,6 +47,51 @@ inline void fill(server::ActiveObjectMgr &mgr, size_t n)
 
 }
 
+template <size_t N>
+void benchInsertObjects(Catch::Benchmark::Chronometer &meter)
+{
+	server::ActiveObjectMgr mgr;
+	
+	meter.measure([&] {
+		fill(mgr, N);
```

Comment:
I think this benchmark is a bit flawed: The lambda passed to `meter.measure` will potentially be called multiple times (see the [Catch2 docs](https://github.com/catchorg/Catch2/blob/devel/docs/benchmarks.md#advanced-benchmarking)). So you would be measuring different things - inserting n objects into an empty AO Manager, inserting n objects into an AO manager with kn objects, etc.

I'm not sure how to fix this. Just moving the AO Manager inside the lambda doesn't cut it, because we don't want to measure time spent clearing.

(This doesn't make much of a difference because your inserts are constant time, though.)

Reply:
Agreed, couldn't come up with a better system. Removing